ingestor/service.go

```diff
@@ -381,6 +391,25 @@ func (s *Service) HandleTransfer(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusAccepted)
 }
 
+func (s *Service) Shutdown() error {
+	if err := s.DisableWrites(); err != nil {
+		logger.Errorf("Failed to disable writes: %s", err.Error())
+		return err
+
+	}
+
+	if err := s.UploadSegments(); err != nil {
+		logger.Errorf("Failed to upload segments: %s", err.Error())
+		return err
+	}
+
+	if err := s.Close(); err != nil {
```

Comment:
I don't think you want to `Close` here because then the pod will just restart and be able to take new writes.  I think it would be better to have the instance stay in this disabled state and signal back that shudown is complete here.  The operator can then see that pod has signaled shutdown is completed and can delete the pod/scale down the statefulset.

That actually reminds me that when we scale down we have to do it in reverse order because of the way that statefulsets works.

Reply:
store.close() is called twice once in DisableWrites() and once here. Restructured the code so that we dont call store.close() in shutdown() and not have to call DisableWrites() either because that one only calls metrics.close() besides store.close()

can you elaborate on the reverse order? I am not sure I follow.